README.md

```diff
@@ -32,44 +30,440 @@ The sync service must first be initialized using the `syncService.init()`. Gener
 });
 ```
 
-Once initialized the syncService can manage multiple `dataset` using the following function:
+##### Managing Datasets
+
+Once initialized the syncService can manage multiple `datasets` using the following function:
 
 ```javascript
 
+var config = {
+  ...
+  datasetId: "workorders"
+  ...
+};
+
 syncService.manage(config.datasetId, null, queryParams);
 
 ```
 
-Checkout a full example [here](https://github.com/feedhenry-staff/wfm-workorder/blob/master/lib/angular/sync-service.js)
 
+### Dataset Topic Subscriptions
 
-#### Topics
+Each `dataset` subscribes to the following topics. For a `dataset` with an ID `datasetid`, the following topics are published:
 
-The following topics are published by the client side of the application:
 
-##### Data Set Topics
+#### Topic Unique Identifiers
 
-The following topics are published by the module for each `dataset`. If the 
+Each of the topics takes an `object` as a parameter. The `topicUid` parameter is an *optional* parameter used to allow the unique identifier to be appended to the `done` and `error` topics published.
 
-###### $fh.sync Client Notifications
+This allows for the scenario where a developer wishes to limit the response of the topic.
+
+E.g
+
+```javascript
+
+//Generate a random number / string
+var topicUid = Math.floor(Math.random() * 100000000);
```

Comment:
Timestamp is just one example. We are just allowing a developer to specify a unique identifier.

I think ultimately, the mediator will be replaced with RxJS https://github.com/Reactive-Extensions/RxJS but that's outside the scope of what I am doing here.